### DISCUSSION TITLE
Custom Route Model Resolution Logic

### DISCUSSION THREAD

**Discussion by krizzdev** [livewire/livewire#3050] [7. Bugs (archived)] [answered]
### Description
Livewire disregards the Route::bind rules in RouteserviceProvider

I need it for the following behavior:

Instead of accessing the Order Item model via the id: 
**/order_items/354726**
I want to access it via the Order Id and the Order Item Position in the order:
**/order_items/182609-1**

### Exact steps to reproduce
I have a Route in web.php:

```php
Route::get('/order_items/{orderItem}', OrderItemViewGeneral::class)->name('order-item-view-general');
```
and the following resolution logic in my RouteServiceProvider:

```php
Route::bind('orderItem', function ($value) {
            $parts = explode('-', $value);
            $orderId = $parts[0];
            $position = $parts[1];
            return OrderItem::where('order_id', $orderId)->where('position', $position)->firstOrFail();
      });
```

If I now access /order_items/182609-1 it will display the livewire component successfully.

But if I add any wire:click or wire:model action to this component it will change the route model key in the browser.

```php
<button wire:click="$toggle('isEditing')" >
          Enable Editing
</button>
```
If I hit this button the **/order_items/182609-1** immediatly changes to **/order_items/354726**

Is there any way of telling livewire to not changing the Model Route Keys ? Or to follow the Route model logic given in the RouteServiceProvider?

**Context**
- Livewire version: [2.4]
- Laravel version: [8.23.1]
- Alpine version: [2.8.1]
- Browser: [Chrome]

**Comment by tshafer** [comment]
I also am struggling with this.

**Comment by LooxisDev** [comment]
Unfortunately we have still the same problem.

**Comment by LooxisDev** [comment, marked as answer]
I found a way with the help of my colleague:

You don't only have to override the 
```
public function resolveRouteBinding($value, $field = null)
{
    return $this->internalAccessor($value)->firstOrFail();
}
```

in the Model. You also have to set the
```
public function getRouteKeyName()
    {
        return 'internal_accessor';
    }
```

Then laravel will use the route key name to map it correctly.
